numba_cuda/numba/cuda/dispatcher.py

```diff
@@ -265,19 +266,33 @@ def maybe_link_nrt(self, link, tgt_ctx, asm):
         if not tgt_ctx.enable_nrt:
```

Comment:
It's hard to check the logic of `maybe_link_nrt()` without being certain what conditions the NRT is supposed to be linked for; I feel like it needn't be this complicated, but that's hard to verify. Can you add some documentation in the form of comments spelling out what conditions are being checked for / why they're being checked?

Reply:
For example - I thought we concluded that we'd require `LinkableCode` instances to explicitly have `nrt` set, so there would be no need to scan the data - the user creating a `LinkableCode` should know whether it's using NRT or not.